service-config/src/main/java/com/palantir/config/service/ServiceConfiguration.java

```diff
@@ -43,4 +43,23 @@
      * A list of service URIs.
      */
     public abstract List<String> uris();
+
+    // hides implementation details
+    public static Builder builder() {
+        return ImmutableServiceConfiguration.builder();
+    }
+
+    // hides implementation details
+    public interface Builder {
+
+        Builder apiToken(BearerToken apiToken);
+
+        Builder security(SslConfiguration security);
```

Comment:
with how things currently are I have to do silly things like https://github.palantir.build/elements/foundry/pull/771/files#diff-f18eac1da998e57c44fff75ab5bc4fdaR67 , also, the method is called "originalServices" instead of "services" which doesn't match. All the more reason to NOT have this overriding/hiding interface